### PR TITLE
Debug `SequenceClassificationRewardModel`

### PR DESCRIPTION
**Pull request by ryokan0123**
The previous implementation assumed that all inputs in the batch had the same length, which caused failures when they did not.
This PR addresses and resolves that issue by adding `padding=True` to `apply_chat_template`.

This PR also specify `return_dict=True` in `apply_chat_template` to explicitly  get `attention_mask` for the input.

### CODE REVIEW

File: tests/core/reward_model/test_sequence_classification.py [hunk 1]
```diff
@@ -10,19 +10,23 @@ def reward_model() -> SequenceClassificationRewardModel:
...
-        )
-        for i in range(num_instances)
+            extra_info={"id": 1},
+        ),
+        RewardBenchInstance(
+            prompt=[{"role": "user", "content": "1 + 1 = ?"}],
+            chosen=[{"role": "assistant", "content": "2"}],
+            rejected=[{"role": "assistant", "content": "3"}],
+            extra_info={"id": 2},
+        ),
```
Comment (butsugiri) [inline tests/core/reward_model/test_sequence_classification.py:25]: Can we assume that these samples have different lengths?

Comment (ryokan0123) [reply tests/core/reward_model/test_sequence_classification.py]: Good point, it wasn't obvious...
I've updated the example to make it clear that the second one is obviously longer than the first

Comment (butsugiri) [reply tests/core/reward_model/test_sequence_classification.py]: Looks great! Thanks for addressing my concerns 👍

**Review by butsugiri — APPROVED** [review]
